webui/react/src/components/VersionChecker.tsx

```diff
@@ -0,0 +1,61 @@
+import Button from 'hew/Button';
+import { useTheme } from 'hew/Theme';
+import { notification } from 'hew/Toast';
+import { useState } from 'react';
+
+import Link from 'components/Link';
+import { paths } from 'routes/utils';
+import { refreshPage } from 'utils/browser';
+
+interface Props {
+  version: string;
+}
+
+const VersionChecker: React.FC<Props> = ({ version }: Props) => {
+  const {
+    themeSettings: { className: themeClass },
+  } = useTheme();
+  const [closed, setClosed] = useState(false);
+  // vi.stubEnv uses string type to set process.env.IS_DEV:
+  const isDev = JSON.parse(process.env.IS_DEV ?? 'false');
+
+  /*
+   * Check to make sure the WebUI version matches the platform version.
+   * Skip this check for development version.
+   */
+  if (!isDev && version !== process.env.VERSION) {
```

Comment:
i think we can put this code back in useEffect

Reply:
prefer not to use `useEffect` unless necessary?